src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -0,0 +1,227 @@
+#include "stdafx.h"
+#include "tpa.h"
+
+// Windows types used by the ExecuteAssembly function
+typedef int32_t HRESULT;
+typedef const char* LPCSTR;
+typedef unsigned int DWORD;
+#define S_OK 0;
+#define E_FAIL -1;
+
+typedef struct CALL_APPLICTION_MAIN_DATA
+{
+    char* applicationBase;
+    char* runtimeDirectory;
+    int argc;
+    char** argv;
+    int exitcode;
+} *PCALL_APPLICATION_MAIN_DATA;
+
+// Prototype of the ExecuteAssembly function from the libcoreclr.do
+typedef HRESULT (*ExecuteAssemblyFunction)(
+                    LPCSTR exePath,
+                    LPCSTR coreClrPath,
+                    LPCSTR appDomainFriendlyName,
+                    int propertyCount,
+                    LPCSTR* propertyKeys,
+                    LPCSTR* propertyValues,
+                    int argc,
+                    LPCSTR* argv,
+                    LPCSTR managedAssemblyPath,
+                    LPCSTR entryPointAssemblyName,
+                    LPCSTR entryPointTypeName,
+                    LPCSTR entryPointMethodsName,
+                    DWORD* exitCode);
+
+bool GetTrustedPlatformAssembliesList(const std::string& szDirectory, bool bNative, std::string& trustedPlatformAssemblies)
+{
+    size_t cTpaAssemblyNames = 0;
+    char** ppszTpaAssemblyNames = NULL;
+
+    CreateTpaBase(&ppszTpaAssemblyNames, &cTpaAssemblyNames, bNative);
+
+    //TODO: The Windows version of this actaully ensures the files are present.  We just fail for native and assume MSIL is present
+    if (bNative)
+    {
+        return false;
+    }
+
+    trustedPlatformAssemblies = std::string();
+
+    for (size_t i = 0; i < cTpaAssemblyNames; i++)
+    {
+        trustedPlatformAssemblies.append(szDirectory);
+        trustedPlatformAssemblies.append("/");
+        trustedPlatformAssemblies.append(ppszTpaAssemblyNames[i]);
+        trustedPlatformAssemblies.append(":");
+    }
+
+    if (ppszTpaAssemblyNames != NULL)
```

Comment:
I think the (unstated) invariant here is that if ppszTpaAssemblyNames is null, then cTpaAssemblyNames is zero.  This same weird pattern is already in dnx.coreclr.  I will add an assert.